packages/ember-glimmer/lib/views/outlet.js

```diff
@@ -25,6 +25,39 @@ class OutletStateReference {
   }
 }
 
+// So this is a relic of the past that SHOULD go away
+// in 3.0. Preferably it is deprecated in the release that
+// follows the Glimmer release.
+export class OrphanedOutletState {
+  constructor(name, inner, orphan) {
+    this.tag = inner.tag;
+    this.inner = inner;
+    this.orphan = orphan;
+    this.orphanState = orphan.get(name).value();
+    this.name = name;
+  }
+
+  value() {
+    let currentOutletState = this.inner.value();
+    let orphanState = this.orphan.get(this.name).value();
+
+    if (!orphanState.wasUsed) {
+      currentOutletState.outlets[this.name] = orphanState;
+      orphanState.wasUsed = true;
+      this.orphanState = orphanState;
+    }
+
+    return currentOutletState;
+  }
+
+  get(name) {
+    let currentOutletState = this.inner.value();
+    currentOutletState.outlets[name] = this.orphanState;
+    this.orphanState.wasUsed = true;
+    return new ChildOutletStateReference(this, name);
+  }
```

Comment:
Do we actually support nested outlets inside an orphan outlet? Do we have a test for it?